cmd/metal-api/internal/service/machine-service.go

```diff
@@ -65,28 +65,29 @@ type machineAllocationSpec struct {
 
 // allocationNetwork is intermediate struct to create machine networks from regular networks during machine allocation
 type allocationNetwork struct {
-	network   *metal.Network
-	ips       []metal.IP
-	auto      bool
-	isPrivate bool
+	network *metal.Network
+	ips     []metal.IP
+	auto    bool
+
+	networkType metal.NetworkType
 }
 
 // allocationNetworkMap is a map of allocationNetworks with the network id as the key
 type allocationNetworkMap map[string]*allocationNetwork
 
-// getPrivateNetwork extracts the private network from an allocationNetworkMap
-func getPrivateNetwork(networks allocationNetworkMap) (*allocationNetwork, error) {
-	var privateNetwork *allocationNetwork
+// getPrivatePrimaryNetwork extracts the private network from an allocationNetworkMap
+func getPrivatePrimaryNetwork(networks allocationNetworkMap) (*allocationNetwork, error) {
+	var privatePrimaryNetwork *allocationNetwork
 	for _, n := range networks {
-		if n.isPrivate {
-			privateNetwork = n
+		if n.networkType.PrivatePrimary {
+			privatePrimaryNetwork = n
 			break
 		}
 	}
-	if privateNetwork == nil {
-		return nil, fmt.Errorf("no private network contained")
+	if privatePrimaryNetwork == nil {
+		return nil, fmt.Errorf("no private primary network contained")
 	}
-	return privateNetwork, nil
+	return privatePrimaryNetwork, nil
```

Comment:
Mh... we now introduced all these nice network types but now we still use the old flags in our algorithms.

What about something like:

```go
func getPrivatePrimaryNetwork(networks allocationNetworkMap) (*allocationNetwork, error) {
	for _, n := range networks {
		if n.networkType == metal.PrivatePrimaryUnshared {
			return n, nil
		}
	}
	for _, n := range networks {
		if n.networkType == metal.PrivatePrimaryShared {
			return n, nil
		}
	}
	return nil, fmt.Errorf("no private primary network contained")
}
``` 
```

Reply:
Mh... I guess, we can even remove this function after #120. So we can leave it as is.